lib/rules/lightness-notation/index.mjs

```diff
@@ -0,0 +1,183 @@
+import valueParser from 'postcss-value-parser';
+
+import declarationValueIndex from '../../utils/declarationValueIndex.mjs';
+import dimensionUnitIsEmpty from '../../utils/dimensionUnitIsEmpty.mjs';
+import getDeclarationValue from '../../utils/getDeclarationValue.mjs';
+import isStandardSyntaxValue from '../../utils/isStandardSyntaxValue.mjs';
+import report from '../../utils/report.mjs';
+import ruleMessages from '../../utils/ruleMessages.mjs';
+import setDeclarationValue from '../../utils/setDeclarationValue.mjs';
+import validateOptions from '../../utils/validateOptions.mjs';
+
+const ruleName = 'lightness-notation';
+
+const messages = ruleMessages(ruleName, {
+	expected: (unfixed, fixed) => `Expected "${unfixed}" to be "${fixed}"`,
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/lightness-notation',
+	fixable: true,
+};
+
+const LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS = new Set(['oklch', 'oklab']);
+const LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS = new Set(['lch', 'lab']);
+const LIGHTNESS_FUNCS = new Set([
+	...LIGHTNESS_ZERO_TO_ONE_RANGE_FUNCS,
+	...LIGHTNESS_ZERO_TO_HUNDRED_RANGE_FUNCS,
+]);
+const HAS_LIGHTNESS_COLOR_FUNC = new RegExp(`\\b(?:${[...LIGHTNESS_FUNCS].join('|')})\\(`, 'i');
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary, _secondaryOptions, context) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, {
+			actual: primary,
+			possible: ['percentage', 'number'],
+		});
+
+		if (!validOptions) return;
+
+		root.walkDecls((decl) => {
+			if (!HAS_LIGHTNESS_COLOR_FUNC.test(decl.value)) return;
+
+			let needsFix = false;
+			const parsedValue = valueParser(getDeclarationValue(decl));
+
+			parsedValue.walk((node) => {
+				if (node.type !== 'function') return;
+
+				if (!LIGHTNESS_FUNCS.has(node.value.toLowerCase())) return;
```

Comment:
[suggest] Can we move the `functionName` declaration at [L66](https://github.com/stylelint/stylelint/pull/7366/files#diff-e55c9791f54478934c2ad8ddd81e2913441c88b5ba16638f72d10ba02214c11aR66) to here?

```suggestion
				const functionName = node.value.toLowerCase();

				if (!LIGHTNESS_FUNCS.has(functionName)) return;
```